examples/gno.land/p/demo/teritori/dao_core/dao_core.gno

```diff
@@ -0,0 +1,188 @@
+	if core.votingModule == nil {
+		panic("voting module factory returned nil")
+	}
+
+	for i, modFactory := range proposalModulesFactories {
+		mod := modFactory(core)
+		if mod == nil {
+			panic("proposal module factory returned nil")
+		}
+		core.proposalModules[i] = dao_interfaces.ActivableProposalModule{
+			Enabled: true,
+			Module:  mod,
+		}
+	}
+
+	// this registry is specific to gno since we can't do dynamic calls
+	core.registry.Register(NewUpdateVotingModuleMessageHandler(core))
+	core.registry.Register(NewUpdateProposalModulesMessageHandler(core))
+	for _, handlerFactory := range messageHandlersFactories {
+		handler := handlerFactory(core)
+		if handler == nil {
+			panic("message handler factory returned nil")
+		}
+		core.registry.Register(handler)
+	}
+
+	return core
+}
+
+// mutations
+
+func (d *daoCore) UpdateVotingModule(newVotingModule dao_interfaces.IVotingModule) {
+	if std.CurrentRealm().Addr() != d.realm.Addr() { // not sure this check necessary since the ownership system should protect against mutation from other realms
+		panic(ErrUnauthorized)
+	}
+
+	d.votingModule = newVotingModule
+}
+
+func (d *daoCore) UpdateProposalModules(toAdd []dao_interfaces.IProposalModule, toDisable []int) {
+	if std.CurrentRealm().Addr() != d.realm.Addr() { // not sure this check necessary since the ownership system should protect against mutation from other realms
+		panic(ErrUnauthorized)
+	}
+
+	for _, module := range toAdd {
+		d.addProposalModule(module)
+	}
+
+	for _, moduleIndex := range toDisable {
+		module := GetProposalModule(d, moduleIndex)
+
+		if !module.Enabled {
+			panic(ErrModuleAlreadyDisabled)
+		}
+		module.Enabled = false
+
+		d.activeProposalModuleCount--
+		if d.activeProposalModuleCount == 0 {
+			panic("no active proposal modules") // this -> `panic(ErrNoActiveProposalModules)` triggers `panic: reflect: reflect.Value.SetString using value obtained using unexported field`
+		}
+	}
+}
+
+// queries
+
+func (d *daoCore) ProposalModules() []dao_interfaces.ActivableProposalModule {
+	return d.proposalModules
+}
+
+func (d *daoCore) VotingModule() dao_interfaces.IVotingModule {
+	return d.votingModule
+}
+
+func (d *daoCore) VotingPowerAtHeight(address std.Address, height int64) uint64 {
+	return d.VotingModule().VotingPowerAtHeight(address, height)
+}
+
+func (d *daoCore) ActiveProposalModuleCount() int {
+	return d.activeProposalModuleCount
+}
+
+func (d *daoCore) Render(path string) string {
+	s := "# DAO Core\n"
+	votingInfo := d.votingModule.Info()
+	s += "## Voting Module: " + votingInfo.String() + "\n"
+	s += d.votingModule.Render("")
+	s += "## Supported Messages:\n"
+	s += d.registry.Render()
+	s += "## Proposal Modules:\n"
+	for i, propMod := range d.proposalModules {
+		if !propMod.Enabled {
+			continue
+		}
+		info := propMod.Module.Info()
+		s += "### #" + strconv.Itoa(i) + ": " + info.String() + "\n"
+		s += propMod.Module.Render("")
+	}
+	return s + "\n"
+}
```

Comment:
When concatenating strings, using `strings.Builder` instead of the `+=` operator can reduce unnecessary memory usage.

```suggestion
func (d *daoCore) Render(path string) string {
    var sb strings.Builder
    sb.WriteString("# DAO Core\n")

    votingInfo := d.votingModule.Info()
    sb.WriteString("## Voting Module: ")
    sb.WriteString(votingInfo.String())
    sb.WriteString("\n")

    sb.WriteString(d.votingModule.Render(""))

    sb.WriteString("## Supported Messages:\n")
    sb.WriteString(d.registry.Render())

    sb.WriteString("## Proposal Modules:\n")
    for i, propMod := range d.proposalModules {
        if !propMod.Enabled {
            continue
        }
        info := propMod.Module.Info()
        sb.WriteString("### #")
        sb.WriteString(strconv.Itoa(i))
        sb.WriteString(": ")
        sb.WriteString(info.String())
        sb.WriteString("\n")
        sb.WriteString(propMod.Module.Render(""))
    }

    sb.WriteString("\n")
    return sb.String()
}
```